### PR TITLE
Add error-prone annotations

### PR DESCRIPTION
**Pull request by lptr**
There are some useful annotations from Error Prone that we could use. Most importantly `@Keep` can be used to mark unused fields and methods if they are needed for some reason.

https://errorprone.info/api/latest/com/google/errorprone/annotations/package-summary.html

### CODE REVIEW

File: platforms/core-execution/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java [hunk 1]
```diff
@@ -442,12 +441,12 @@ private SnapshotHierarchy withWatcherChangeErrorHandling(SnapshotHierarchy curre
...
         } catch (Exception ex) {
-            logWatchingError(ex, FILE_WATCHING_ERROR_MESSAGE_DURING_BUILD, null);
+            logWatchingError(ex, FILE_WATCHING_ERROR_MESSAGE_DURING_BUILD);
             return stopWatchingAndInvalidateHierarchyAfterError(currentRoot);
         }
     }
 
-    private void logWatchingError(Exception exception, String fileWatchingErrorMessage, @Nullable WatchMode watchMode) {
+    private void logWatchingError(Exception exception, String fileWatchingErrorMessage) {
         if (exception instanceof InotifyInstanceLimitTooLowException) {
```
Comment (reinsch82) [inline platforms/core-execution/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java:450]: 🎉

**Review by reinsch82 — APPROVED** [review]
LGTM

**Review by asodja — APPROVED** [review]
LGTM

**Comment by lptr** [comment]
@bot-gradle merge this please.

**Comment by lptr** [comment]
Unfortunately we can't merge this, because of dependency validation. Added an issue to add this later: https://github.com/gradle/gradle/issues/27953

**Pull request closed without merging**